### PR TITLE
feat(route): add 聯合新聞網

### LINKED ISSUE #10495: 聯合新聞網
### 类型

传统媒体

### 网站地址

https://udn.com/news/index

### 网站描述

新聞資訊

### 需要生成什么内容？

即時列表/不分類RSS  https://udn.com/news/breaknews/1/99#breaknews

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by miles170**
## 该 PR 相关 Issue / Involved issue

Close #10495

## 完整路由地址 / Example for the proposed route(s)

```routes
/udn/news/breakingnews/0
/udn/news/breakingnews/1
/udn/news/breakingnews/2
/udn/news/breakingnews/3
/udn/news/breakingnews/4
/udn/news/breakingnews/5
/udn/news/breakingnews/6
/udn/news/breakingnews/7
/udn/news/breakingnews/8
/udn/news/breakingnews/9
/udn/news/breakingnews/11
/udn/news/breakingnews/12
/udn/news/breakingnews/13
/udn/news/breakingnews/99
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/udn/breaking-news.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const { art } = require('@/utils/render');
+const path = require('path');
+const timezone = require('@/utils/timezone');
+const cheerio = require('cheerio');
+
+module.exports = async (ctx) => {
+    const id = ctx.params.id;
+    const link = `/news/breaknews/1/${id}#breaknews`;
+    const name = await getLinkName(ctx, link);
+    const rootUrl = 'https://udn.com/';
```
Comment (TonyRL) [inline lib/v2/udn/breaking-news.js:12]:
```suggestion
    const rootUrl = 'https://udn.com';
```
So response url and `link` won't becomes https://udn.com//api/more......

File: lib/v2/udn/breaking-news.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    const response = await got(`${rootUrl}/api/more?page=1&channelId=1&cate_id=${id}&type=breaknews`);
+    const items = await Promise.all(
+        response.data.lists.map(async (item) => {
+            const link = item.titleLink.indexOf('http://') === 0 || item.titleLink.indexOf('https://') === 0 ? item.titleLink : `${rootUrl}${item.titleLink}`;
```
Comment (TonyRL) [inline lib/v2/udn/breaking-news.js:16]: Consider removing the `from` search params from url

File: lib/v2/udn/breaking-news.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+            const { author, description } = await ctx.cache.tryGet(link, async () => {
+                const result = await got(link);
+                const $ = cheerio.load(result.data);
+
+                return {
+                    author: $('.article-content__author a').text().trim(),
+                    description: art(path.join(__dirname, 'templates/description.art'), {
+                        description: $('.article-content__editor  p').text().trim(),
```
Comment (TonyRL) [inline lib/v2/udn/breaking-news.js:25]:
```suggestion
                        description: $('.article-content__editor').html(),
```
Why parse text only? Some inline pictures will be skipped in that way like [this one](https://udn.com/news/story/7270/6575991).

Please add `$('.article-body__editor').html()` also in order to parse part of the content that blocked behind paywall like [this one](https://udn.com/news/story/6812/6576003).

File: lib/v2/udn/templates/description.art [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<p>{{ description }}</p>
+{{ if image }}
+<img src="{{ image }}">
+{{ /if }}
+{{ author }}
```
Comment (TonyRL) [inline lib/v2/udn/templates/description.art:5]: Unused attribute
```suggestion
```

File: lib/v2/udn/breaking-news.js (hunk 2 above)
Comment (miles170) [reply lib/v2/udn/breaking-news.js:16]: There is no `from` search params

File: lib/v2/udn/breaking-news.js (hunk 3 above)
Comment (miles170) [reply lib/v2/udn/breaking-news.js:25]: Should we cleanup the style element from the HTML output?

File: lib/v2/udn/breaking-news.js (hunk 2 above)
Comment (TonyRL) [reply lib/v2/udn/breaking-news.js:16]: image

**Comment by miles170** [comment]
[#a18edf4](https://github.com/DIYgod/RSSHub/pull/10654/commits/a18edf42a135dd344ea27b59e8c7ae8f88ed0bfa) fix the VIP article that is not parsed properly

e.g. https://udn.com/news/story/7331/6576320

File: lib/v2/udn/breaking-news.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+const got = require('@/utils/got');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+const cheerio = require('cheerio');
+const { URL } = require('url');
```
Comment (TonyRL) [inline lib/v2/udn/breaking-news.js:5]: No need to import `URL`
```suggestion
const { art } = require('@/utils/render');
const path = require('path');
```

File: lib/v2/udn/breaking-news.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            });
+
+            return {
+                title: item.title,
+                author,
+                description,
+                pubDate: timezone(parseDate(item.time.date, 'YYYY-MM-DD HH:mm'), +8),
+                link,
+            };
+        })
```
Comment (TonyRL) [inline lib/v2/udn/breaking-news.js:62]: The webpage has embedded JSON which you can make you of.
You can put other properties into cache as well.
```suggestion
        response.data.lists.map((item) => {
            let link = item.titleLink.indexOf('http://') === 0 || item.titleLink.indexOf('https://') === 0 ? item.titleLink : `${rootUrl}${item.titleLink}`;
            const linkUrl = new URL(link);
            // cleanup query paramter
            linkUrl.query = linkUrl.search = '';
            link = linkUrl.toString();

            return ctx.cache.tryGet(link, async () => {
                let result = await got(link);
                // VIP article requires redirection
                // e.g. https://udn.com/news/story/7331/6576320
                const vip = result.data.match(/<script language=javascript>window\.location\.href="(https?:\/\/[^"]+")/);
                if (vip) {
                    result = await got(vip[1]);
                }
                const $ = cheerio.load(result.data);
                const metadata = $('script[type="application/ld+json"]').eq(0).text().trim();
                const data = metadata.startsWith('[') ? JSON.parse(metadata)[0] : JSON.parse(metadata);

                // e.g. https://udn.com/news/story/7001/6576479
                const content = $('.article-content__editor');
                // e.g. https://udn.com/news/story/7240/6576424
                const body = $('.article-body__editor');

                let description = '';
                if (data.image) {
                    description += art(path.join(__dirname, 'templates/figure.art'), {
                        src: data.image.contentUrl,
                        alt: data.image.name,
                    });
                }
                if (content.length) {
                    description += content.html();
                } else if (body.length) {
                    description += body.html();
                }

                return {
                    title: item.title,
                    author: data.author.name,
                    description,
                    pubDate: timezone(parseDate(item.time.date, 'YYYY-MM-DD HH:mm'), +8),
                    category: [data.articleSelection, ...data.keywords.split(',')],
                    link,
                };
            });
        })
```
```html
<figure>
<picture><img src={{ src }} alt={{ alt }}></picture>
<figurecaption>{{ alt }}</figurecaption>
</figure>
```

**Pull request merged**